sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -680,38 +672,38 @@ impl TransactionBuilder {
         })
     }
 
-    // Inputs need to be sorted before signing, because the reference unlock conditions can only reference a lower index
-    pub(crate) fn sort_input_signing_data(
-        mut inputs: Vec<InputSigningData>,
-        commitment_slot_index: SlotIndex,
-        committable_age_range: CommittableAgeRange,
-    ) -> Result<Vec<InputSigningData>, TransactionBuilderError> {
-        // initially sort by output to make it deterministic
-        // TODO: rethink this, we only need it deterministic for tests, for the protocol it doesn't matter, also there
-        // might be a more efficient way to do this
-        inputs.sort_by_key(|i| i.output.pack_to_vec());
-        // filter for ed25519 address first
-        let (mut sorted_inputs, account_nft_address_inputs): (Vec<InputSigningData>, Vec<InputSigningData>) =
-            inputs.into_iter().partition(|input_signing_data| {
+    fn insert_input_sorted(&mut self, input: InputSigningData) {
+        let input_required_address = input
+            .output
+            .required_address(
+                self.latest_slot_commitment_id.slot_index(),
+                self.protocol_parameters.committable_age_range(),
+            )
+            // PANIC: safe to unwrap as non basic/account/foundry/nft outputs are already filtered out.
```

Comment:
Regarding the comment, these outputs aren't filtered out here? Same in L 703

Reply:
good point

Reply:
though they were never correct, so this isn't any different

Reply:
They were correct when we had the treasury output, but not updated when we removed it
```Rust
    pub fn unlock_conditions(&self) -> Option<&UnlockConditions> {
        match self {
            Self::Basic(output) => Some(output.unlock_conditions()),
            Self::Account(output) => Some(output.unlock_conditions()),
            Self::Anchor(output) => Some(output.unlock_conditions()),
            Self::Foundry(output) => Some(output.unlock_conditions()),
            Self::Nft(output) => Some(output.unlock_conditions()),
            Self::Delegation(output) => Some(output.unlock_conditions()),
        }
    }
```
Should just remove the option here I guess

Reply:
actually, they are filtered out of the available outputs, so this is true

Reply:
Delegation outputs can't be there?

Reply:
No, but they are safe to unwrap at least